### PR TITLE
Fix `ends_at_present_flag` null pointer error

### LINKED ISSUE #164: Fix null pointer checking ends_at_present_flag while validating echo10 metadata
**Describe the bug**
Our test framework is trying to validate an ECHO10 record and getting a null pointer when checking ends_at_present_flag, see diff below:

```
diff --git a/pyQuARC/code/custom_validator.py b/pyQuARC/code/custom_validator.py
index 3171bcd..08e3add 100644
--- a/pyQuARC/code/custom_validator.py
+++ b/pyQuARC/code/custom_validator.py
@@ -130,8 +130,8 @@ class CustomValidator(BaseValidator):
     def user_services_check(first_name, middle_name, last_name):
         return {
             "valid": not (
                 (not middle_name or (middle_name.lower() == 'null'))
             ),
             "value": f'{first_name} {middle_name} {last_name}'
@@ -159,9 +159,9 @@ class CustomValidator(BaseValidator):
         # Logic: https://github.com/NASA-IMPACT/pyQuARC/issues/61
         validity = True
         if collection_state.upper() in ["ACTIVE", "IN WORK"]:
-            validity = (not bool(ending_date_time)) and ends_at_present_flag.lower() == "true"
+            validity = (not bool(ending_date_time)) and (ends_at_present_
[failed_test.txt](https://github.com/NASA-IMPACT/pyQuARC/files/8116047/failed_test.txt)
flag and ends_at_present_flag.lower() == "true")
         elif collection_state.upper() == "COMPLETE":
-            validity = bool(ending_date_time) and (not bool(ends_at_present_flag) or ends_at_present_flag.lower() == "false")
+            validity = bool(ending_date_time) and (not bool(ends_at_present_flag) or (ends_at_present_flag and ends_at_present_flag.lower() == "false"))
         else:
             validity = False
         return {

```

### PR DESCRIPTION
**Pull request by slesaad**
Fixes #164
Updates version and changelog

### CODE REVIEW

File: pyQuARC/code/custom_validator.py [hunk 1]
```diff
@@ -159,9 +159,9 @@ def collection_progress_consistency_check(collection_state, ends_at_present_flag
         # Logic: https://github.com/NASA-IMPACT/pyQuARC/issues/61
         validity = True
         if collection_state.upper() in ["ACTIVE", "IN WORK"]:
-            validity = (not bool(ending_date_time)) and ends_at_present_flag.lower() == "true"
+            validity = (not bool(ending_date_time)) and ends_at_present_flag and ends_at_present_flag.lower() == "true"
         elif collection_state.upper() == "COMPLETE":
-            validity = bool(ending_date_time) and (not bool(ends_at_present_flag) or ends_at_present_flag.lower() == "false")
+            validity = bool(ending_date_time) and (not bool(ends_at_present_flag) or (ends_at_present_flag and ends_at_present_flag.lower() == "false"))
```
Comment (xhagrg) [inline pyQuARC/code/custom_validator.py:164]: Re-Use Variables?

File: pyQuARC/code/custom_validator.py [hunk 2]
```diff
@@ -158,10 +158,19 @@ def boolean_check(field_value):
     def collection_progress_consistency_check(collection_state, ends_at_present_flag, ending_date_time):
         # Logic: https://github.com/NASA-IMPACT/pyQuARC/issues/61
         validity = True
```
Comment (xhagrg) [inline pyQuARC/code/custom_validator.py:160]: Start with validity = False. You can then remove the else condition.

File: pyQuARC/code/custom_validator.py [hunk 3]
```diff
@@ -158,10 +158,19 @@ def boolean_check(field_value):
...
-        if collection_state.upper() in ["ACTIVE", "IN WORK"]:
-            validity = (not bool(ending_date_time)) and ends_at_present_flag and ends_at_present_flag.lower() == "true"
-        elif collection_state.upper() == "COMPLETE":
-            validity = bool(ending_date_time) and (not bool(ends_at_present_flag) or (ends_at_present_flag and ends_at_present_flag.lower() == "false"))
+        collection_state = collection_state.upper()
+        ending_date_time_exists = bool(ending_date_time)
+        ends_at_present_flag_exists = bool(ends_at_present_flag)
+        ends_at_present_flag = str(ends_at_present_flag_exists) if ends_at_present_flag_exists else None
```
[the remark this answers is not in the document]
Comment (slesaad) [reply pyQuARC/code/custom_validator.py]: meant to convert to lowercase, my bad

[the remark this answers is not in the document]
Comment (slesaad) [reply pyQuARC/code/custom_validator.py]: [e5d09f1](https://github.com/NASA-IMPACT/pyQuARC/pull/165/commits/e5d09f15e2b27d45e14b909755abe4e4b993374c)

**Review by xhagrg — APPROVED** [review]
LGTM.

**Pull request merged**